tools-v2/pkg/cli/command/curvebs/delete/volume/recover/recover.go

```diff
@@ -71,77 +72,81 @@ func (rCmd *RecoverCmd) Init(cmd *cobra.Command, args []string) error {
+		var payload struct {
+			snapshotutil.Response
+			TaskInfos  snapshotutil.TaskInfos `json:"TaskInfos"`
+			TotalCount int                    `json:"TotalCount"`
 		}
-		if err := json.Unmarshal([]byte(result), &resp); err != nil {
+		if err := json.Unmarshal([]byte(result), &payload); err != nil {
 			return err
 		}
-		if resp.Code != cobrautil.ResultSuccess {
-			return fmt.Errorf("get clone list fail, error code: %s", resp.Code)
+		if payload.Code != snapshotutil.ResultSuccess {
+			return fmt.Errorf("get clone list fail, requestId: %s, code: %s, message: %s", payload.RequestId, payload.Code, payload.Message)
 		}
-		if len(resp.TaskInfos) == 0 {
+		if len(payload.TaskInfos) == 0 {
 			break
 		} else {
-			records = append(records, resp.TaskInfos...)
-			params[cobrautil.QueryOffset] = params[cobrautil.QueryOffset].(int) + params[cobrautil.QueryLimit].(int)
+			records = append(records, payload.TaskInfos...)
+			params[snapshotutil.QueryOffset] = params[snapshotutil.QueryOffset].(int) + params[snapshotutil.QueryLimit].(int)
 		}
 	}
 
 	wg := sync.WaitGroup{}
 	for _, item := range records {
 		wg.Add(1)
-		go func(item map[string]string) {
+		go func(clone snapshotutil.TaskInfo) {
 			defer wg.Done()
+			result := cobrautil.ROW_VALUE_SUCCESS
+			reason := ""
 			params := map[string]any{
-				cobrautil.QueryAction: cobrautil.ActionCleanCloneTask,
-				cobrautil.QueryUser:   item["User"],
-				cobrautil.QueryUUID:   item["UUID"],
+				snapshotutil.QueryAction: snapshotutil.ActionCleanCloneTask,
+				snapshotutil.QueryUser:   clone.User,
+				snapshotutil.QueryUUID:   clone.UUID,
 			}
-			subUri := cobrautil.NewSnapshotQuerySubUri(params)
+			subUri := snapshotutil.NewSnapshotQuerySubUri(params)
 			metric := basecmd.NewMetric(rCmd.snapshotAddrs, subUri, rCmd.timeout)
-			result, err := basecmd.QueryMetric(metric)
+			retStr, err := basecmd.QueryMetric(metric)
 			if err.TypeCode() != cmderror.CODE_SUCCESS {
-				item["Result"] = cobrautil.ROW_VALUE_FAILED
+				result = cobrautil.ROW_VALUE_FAILED
+				reason = err.ToError().Error()
 			} else {
-				payload := map[string]any{}
-				if err := json.Unmarshal([]byte(result), &payload); err != nil {
-					item["Result"] = cobrautil.ROW_VALUE_FAILED
+				payload := snapshotutil.Response{}
+				if err := json.Unmarshal([]byte(retStr), &payload); err != nil {
+					result = cobrautil.ROW_VALUE_FAILED
+					reason = err.Error()
 				} else {
-					if payload[cobrautil.ResultCode] != cobrautil.ResultSuccess {
-						item["Result"] = cobrautil.ROW_VALUE_FAILED
+					if payload.Code != snapshotutil.ResultSuccess {
+						result = cobrautil.ROW_VALUE_FAILED
+						reason = payload.Message
 					}
 				}
 			}
-			item["Result"] = cobrautil.ROW_VALUE_SUCCESS
-			rCmd.TableNew.Append([]string{item["User"], item["Src"], item["UUID"], item["File"], item["Result"]})
+			rCmd.TableNew.Append([]string{clone.User, clone.Src, clone.UUID, clone.File, result, reason})
```

Comment:
Is it appropriate to write it here? The original set is to avoid having to adjust the order of the headers here as well.